src/fhirtypes/common.ts

```diff
@@ -198,6 +198,20 @@ export function setPropertyOnInstance(
           if (current[key] == null) current[key] = {};
           current = current[key];
         } else {
+          // Check if the instance  already has the element being defined
+          if (current.hasOwnProperty([pathPart.base])) {
```

Comment:
I'm surprised this works, but it clearly does. The [MDN page for hasOwnProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) doesn't mention taking an array parameter, though, so I would suggest passing the argument as a string rather than an array.

Reply:
Looking at this now, I'm actually surprised that it works as well haha. The brackets were a holdover from a previous implementation that I somehow glanced over a few times before I pushed this PR. I'll remove this in an edit, and then combine the three conditionals as well.